File: Movieweb/src/models/Movie.ts
import type { Director } from "./Director";

export class Movie {
    id: number;
    name: string;
    releaseYear?: string;
    gender?: string;
    duration?: string; 
    fkdirector?: number;
    fkdirectorNavigation?: Director;

    constructor() {
        this.id = 0;
        this.name = "";
    }
}

File: MovieAPI/MovieAPI/Models/Movie.cs
using System;
using System.Collections.Generic;

namespace MoviesAPI.Models;

public partial class Movie
{
    public int Id { get; set; }

    public string Name { get; set; } = null!;

    public DateOnly? ReleaseYear { get; set; }

    public string? Gender { get; set; }

    public TimeOnly? Duration { get; set; }

    public int? Fkdirector { get; set; }

    public virtual Director? FkdirectorNavigation { get; set; }
}
